tests/unit/structures/test_workflow.py

```diff
@@ -294,28 +294,28 @@ def test_run_topology_1_declarative_mixed(self):
         self._validate_topology_1(workflow)
 
     def test_run_topology_1_imperative_parents(self):
+        workflow = Workflow(prompt_driver=MockPromptDriver())
         task1 = PromptTask("test1", id="task1")
-        task2 = PromptTask("test2", id="task2")
-        task3 = PromptTask("test3", id="task3")
-        task4 = PromptTask("test4", id="task4")
+        task2 = PromptTask("test2", id="task2", structure=workflow)
+        task3 = PromptTask("test3", id="task3", structure=workflow)
+        task4 = PromptTask("test4", id="task4", structure=workflow)
         task2.add_parent(task1)
         task3.add_parent("task1")
         task4.add_parents([task2, "task3"])
-        workflow = Workflow(prompt_driver=MockPromptDriver(), tasks=[task1, task2, task3, task4])
```

Comment:
The old method should still work, right? Maybe you should add new tests instead of modifying them?